zookeeper/src/main/java/com/linecorp/armeria/common/zookeeper/DefaultNodeValueCodec.java

```diff
@@ -98,7 +98,13 @@ public Endpoint decode(String segment) {
 
     @Override
     public byte[] encode(Endpoint endpoint) {
-        return (endpoint.host() + fieldDelimiter + endpoint.port() + fieldDelimiter + endpoint.weight())
+        final String ipAddressOrHostname;
+        if (endpoint.hasIpAddr()) {
+            ipAddressOrHostname = endpoint.ipAddr();
+        } else {
+            ipAddressOrHostname = endpoint.host();
+        }
+        return (ipAddressOrHostname + fieldDelimiter + endpoint.port() + fieldDelimiter + endpoint.weight())
```

Comment:
This can break things, especially when a client has to send the host name.

- `Host` or `:authority` headers
- TLS SNI